packager/mpd/base/adaptation_set.cc

```diff
@@ -274,6 +274,14 @@ base::Optional<xml::XmlNode> AdaptationSet::GetXml() {
       return base::nullopt;
     }
   }
+  if (subsegment_start_with_sap_ == 1) {
```

Comment:
could you please add some tests and potentially since this is for spec compliance all the existing golden tests will also be regenrated once you can run those tests as well. 

assuming this is for audio only and hence the subsegment check for just value == 1 ?